packages/orbit-components/src/Box/index.tsx

```diff
@@ -46,23 +45,319 @@ const Box = React.forwardRef<HTMLDivElement, Props>(
+        if (viewportProps == null) {
+          return null;
+        }
+
+        const {
+          display: vpDisplay,
+          wrap: vpWrap,
+          direction: vpDirection,
+          position: vpPosition,
+          align: vpAlign,
+          justify: vpJustify,
+          textAlign: vpTextAlign,
+          elevation: vpElevation,
+          borderRadius: vpBorderRadius,
+          overflow: vpOverflow,
+          color: vpColor,
+          background: vpBackground,
+          padding: vpPadding,
+          margin: vpMargin,
+        } = viewportProps;
+
+        return [
+          vpDisplay && displayClasses[viewport][vpDisplay],
+          vpWrap && wrapClasses[viewport][vpWrap],
+          vpDirection && directionClasses[viewport][vpDirection],
+          vpPosition && positionClasses[viewport][vpPosition],
+          vpAlign && alignClasses[viewport][vpAlign],
+          vpJustify && justifyClasses[viewport][vpJustify],
+          vpTextAlign && textAlignClasses[viewport][vpTextAlign],
+          vpElevation && shadowClasses[viewport][vpElevation],
+          vpBorderRadius && borderRadiusClasses[viewport][vpBorderRadius],
+          vpOverflow && overflowClasses[viewport][vpOverflow],
+          vpColor && colorClasses[viewport][vpColor],
+          vpBackground && backgroundColorClasses[viewport][vpBackground],
+          vpPadding && typeof vpPadding === "string" && paddingClasses[viewport][vpPadding],
+          vpPadding &&
+            typeof vpPadding === "object" &&
+            vpPadding.top &&
+            paddingTopClasses[viewport][vpPadding.top],
+          vpPadding &&
+            typeof vpPadding === "object" &&
+            vpPadding.right &&
+            paddingRightClasses[viewport][vpPadding.right],
+          vpPadding &&
+            typeof vpPadding === "object" &&
+            vpPadding.bottom &&
+            paddingBottomClasses[viewport][vpPadding.bottom],
+          vpPadding &&
+            typeof vpPadding === "object" &&
+            vpPadding.left &&
+            paddingLeftClasses[viewport][vpPadding.left],
+          vpMargin && typeof vpMargin === "string" && marginClasses[viewport][vpMargin],
+          vpMargin &&
+            typeof vpMargin === "object" &&
+            vpMargin.top &&
+            marginTopClasses[viewport][vpMargin.top],
+          vpMargin &&
+            typeof vpMargin === "object" &&
+            vpMargin.right &&
+            marginRightClasses[viewport][vpMargin.right],
+          vpMargin &&
+            typeof vpMargin === "object" &&
+            vpMargin.bottom &&
+            marginBottomClasses[viewport][vpMargin.bottom],
+          vpMargin &&
+            typeof vpMargin === "object" &&
+            vpMargin.left &&
+            marginLeftClasses[viewport][vpMargin.left],
+        ];
+      });
+    }, [mediumMobile, largeMobile, tablet, desktop, largeDesktop]);
+
+    const vars = {
+      "--box-shrink": shrink,
+      "--ld-box-shrink": largeDesktop?.shrink,
```

Comment:
I thought we decided to use variables when it's hard to actually have a large object of classNames for such properties as max-height, width and etc. In case like shrink, grow -  I do not see a problem to have it with classNames, it later can be reused for Stack too. :)

Reply:
yes, so for now, I think we can use it as separate classes because of differences in components API. As you say, it's uncommon to use values greater than 1. In the tracking, nobody uses values > 1 (18 occurrences in total I found). 

What if we will unify that in a common folder to be a boolean and in Box it will be conditionally applied, like: 

```
typeof shrink !== "undefined" && shrinkClasses[Boolean(shrink)]
```

Reply:
That is still a breaking change on whoever is using it with a different value. It has no impact on our teams as far as our tracking tool can detect. We can accept it and consider it to be non-impactful on our primary users, but it should still be documented as a BC if we do it like that.

Or we can leave it as is and revisit when we plan to unify the Box and Stack APIs

Reply:
No problem. Document it as breaking change then ;)

Reply:
Right, I meant the string value true/false or we can use numbers like 0 and 1. But anyway, I hope you got what I meant 🙂 

My point was to not limit the feature because nobody uses greater numbers and perhaps won't, so it's not limiting imo, but to unify that logic on our side. I will leave the solution up to you then, it's not critical and I won't insist on that 🙂